src/main/java/ezschedule/logic/commands/EditCommand.java

```diff
@@ -92,10 +93,12 @@ public CommandResult execute(Model model) throws CommandException {
         Event eventToEdit = lastShownList.get(index.getZeroBased());
         Event editedEvent = createEditedEvent(eventToEdit, editEventDescriptor);
 
-        if (!eventToEdit.equals(editedEvent) && model.hasEvent(editedEvent)) {
-            throw new CommandException(MESSAGE_DUPLICATE_EVENT);
-        } else if (editedEvent.getEndTime().isBefore(editedEvent.getStartTime())) {
+        if (editedEvent.getEndTime().isBefore(editedEvent.getStartTime())) {
             throw new CommandException(MESSAGE_EVENT_END_TIME_EARLIER_THAN_START_TIME);
+        } else if (!eventToEdit.equals(editedEvent) && model.hasEvent(editedEvent)) {
```

Comment:
wrt `!eventToEdit.equals(editedEvent)`, assuming that checks ensured that at least something is edited, then this statement will always be `true`. Possible to remove to improve code quality?

Reply:
Nope. Causes 1 of the test to fail with the change